EthersJs/accelerating-development-with-ethersjs.md

```diff
@@ -0,0 +1,242 @@
+
+```
+import { Component, OnInit } from '@angular/core';
+import { ethers } from 'ethers';
+import { FormBuilder, FormGroup, NgForm } from '@angular/forms'; 
+
+@Component({
+  selector: 'app-wallet',
+  templateUrl: './wallet.component.html',
+  styleUrls: ['./wallet.component.scss']
+})
+export class WalletComponent implements OnInit {
+
+    publickey: string;
+    privatekey: string;
+    transactionForm: FormGroup;
+    toAddress: string;
+    etherAmount: string;
+    wallet: any;
+    provider: any;
+
+  constructor(private fb: FormBuilder) { 
+      this.transactionForm = fb.group({
+          'toAddress': [null],
+          'etherAmount': [null]
+      });
+  }
+
+  ngOnInit() {
+      this.provider = ethers.getDefaultProvider('homestead');;
+  }
+
+  onSubmit() {
+      const randomWallet = ethers.Wallet.createRandom();
+      this.publickey = randomWallet.address;
+      this.privatekey = randomWallet.privateKey;
+      this.wallet = new ethers.Wallet(this.privatekey, this.provider)
+  }
+
+}
+```
+
+In order to send a transaction, Ethersjs provides a simple sendTransaction method for all wallets. Add a sendTransaction method to the wallet class, inputting the form.
+
+```
+sendTransaction(form: NgForm) {
+    let transaction = {
+      to: form.toAddress,
+      value: ethers.utils.parseEther(form.etherAmount)
+    }
+
+    this.wallet.sendTransaction(transaction)
+        .then((tx) => {
+            console.log(tx);
+        })
+}
+```
+
+We first create the transaction object, giving where the transaction is going in the `to` field. `value` denotes how much ether, default in the units Wei, is being sent to the address mentioned. We use the parseEther util provided by ethers to easily convert from Ether to Wei. After creating the transaction object, we use our wallet we created before to send the transaction. In this implementation, the console logs the transaction receipt.
+
+Signing a message in a transaction is very useful for a number of reasons. A signed message can be used to verify that the owner is the address provided, and this allows smart contracts and other entities to verify the truth in a trustless system. To sign a message using Ethers, you use the signMessage function on your message.
+
+```
+wallet.signMessage(message);
+```
+
+## Interacting with Smart Contracts
+
+One of the many novelties of Ethereum is the creation and use of smart contracts on the blockchain. Dapp development relies on easy interaction with smart contracts, and Ethers has an easy to use solution.
+
+Take for example this very simple [Sample Contract](https://ropsten.etherscan.io/address/0x8a32989b65186d3596251d7d7c8a427a26669354#code). In this contract, we store variables by adding it to the blockchain and can read all currently stored variables. Interacting with this contract with Ethers, we first create the ABI for it, found [here](https://ropsten.etherscan.io/address/0x8a32989b65186d3596251d7d7c8a427a26669354#code).
```

Comment:
```suggestion
Take for example this [Sample Contract](https://ropsten.etherscan.io/address/0x8a32989b65186d3596251d7d7c8a427a26669354#code). In this contract, we store variables by adding it to the blockchain and can read all currently stored variables. Interacting with this contract with Ethers, we first create the ABI for it, found [here](https://ropsten.etherscan.io/address/0x8a32989b65186d3596251d7d7c8a427a26669354#code).
```

Reply:
And why do we need to do this? Might be good to explain the why before diving straight into the how.

Reply:
Updated with a little more explanation. Please let me know if this suffices.